src/stepFunctions/commands/publishStateMachine.ts

```diff
@@ -30,6 +30,10 @@ export async function publishStateMachine(awsContext: AwsContext, outputChannel:
         throw new Error('Could not get active text editor for state machine definition')
     }
 
+    if (textDocument.languageId == 'yasl') {
+        throw new Error('Cannot publish state machine from Amazon States Language in YAML')
```

Comment:
Is this messaged surfaced to the user? Will users be conscious of this beforehand?

Reply:
Seems like this should be logged, and a localized error shown to the user somewhere (whether it is here, or catching and surfacing somewhere up the call chain). This was missed when reviewing line 30 at some time earlier.

Seems like StepFunctionsCodeLensProvider could be revised to not show the publish codelens for yaml files as well.

Reply:
Good idea, I will change it so that the publish codelens option will not appear on yasl files.